src/shared_memory.rs

```diff
@@ -0,0 +1,12 @@
+pub trait ShareableMemory {
+    fn driver_number(&self) -> usize;
+
+    fn allow_number(&self) -> usize;
+
+    fn to_bytes(&self) -> &[u8];
+}
+
+pub struct SharedMemoryHandle<SM: ShareableMemory> {
```

Comment:
What about `MemoryShare` or `SharedMemory` in analogy to `SubscribableCallback` and `CallbackSubscription`? To me `SomethingHandle` has no clear meaning.

Reply:
Right, I'll change that.